tests/integration/dualstack/common/apps.go

```diff
@@ -0,0 +1,125 @@
+//go:build integ
+// +build integ
+
+//  Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+var prom prometheus.Instance
+
+const (
+	Unknown = iota
+	IPv4
+	IPv6
+)
+
+type EchoDeployments struct {
+	// Service Name
+	ServiceName string
+	// IPFamily of echo app only if single-stack IPv4 or IPv6
+	IPFamily int
+	// Namespace echo apps will be deployed
+	Namespace namespace.Instance
+	// Standard echo app to be used by tests
+	EchoPod echo.Instance
+}
+
+type DeploymentConfig struct {
+	Name           string
+	Service        string
+	IPFamily       string
+	IPFamilyPolicy string
+}
+
+var deploymentConfigs = []DeploymentConfig{
+	{Name: "ipv4", Service: "echo-ipv4", IPFamily: "IPv4", IPFamilyPolicy: "SingleStack"},
+	{Name: "ipv6", Service: "echo-ipv6", IPFamily: "IPv6", IPFamilyPolicy: "SingleStack"},
+	{Name: "dual-ipv4", Service: "echo-dual-ipv4", IPFamily: "IPv4, IPv6", IPFamilyPolicy: "RequireDualStack"},
+	{Name: "dual-ipv6", Service: "echo-dual-ipv6", IPFamily: "IPv6, IPv4", IPFamilyPolicy: "RequireDualStack"},
+}
+
+var EchoPorts = []echo.Port{
```

Comment:
Can we re-use (a portion) of common echo deployments from other suites?

Reply:
I'll do that here. We did have our own version for other tests that we haven't upstreamed yet but for this case it makes sense to delay adding this.